DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -80,93 +125,124 @@ function Test-TargetResource
+}
 
 #######################################################################################
-#  Helper function that validates the Server Address properties. If the switch parameter
-# "Apply" is set, then it will set the properties after a test
+#  Helper functions
 #######################################################################################
-function Test-Properties
-{
+function Test-ResourceProperty {
+    # Function will check the Address details are valid and do not conflict with
+    # Address family. Ensures interface exists.
+    # If any problems are detected an exception will be thrown.
+    [CmdLetBinding()]
     param
     (
-        [Parameter(Mandatory)]
-        [ValidateNotNullOrEmpty()]
-        [String[]]$Address,
+        [String]$Address,
 
         [Parameter(Mandatory)]
         [ValidateNotNullOrEmpty()]
         [String]$InterfaceAlias,
 
-        [Parameter(Mandatory)]
-        [ValidateSet("IPv4", "IPv6")]
-        [String]$AddressFamily,
-
-        [Switch]$Apply
+        [ValidateSet('IPv4', 'IPv6')]
+        [String]$AddressFamily = 'IPv4'
     )
-    $sa =$Address
-    $sa | Foreach-Object {
-             if(!([System.Net.Ipaddress]::TryParse($_, [ref]0)))
-             {
-                 throw "Server Address *$_* is not in the correct format. Please correct the Address in the configuration and try again"
-             }
-             if (([System.Net.IPAddress]$_).AddressFamily.ToString() -eq [System.Net.Sockets.AddressFamily]::InterNetwork.ToString())
-             {
-                if ($AddressFamily -ne "IPv4")
-                {
-                    throw "Server address *$_* is in IPv4 format, which does not match server address family $AddressFamily. Please correct either of them in the configuration and try again"
-                }
-             }
-             else
-             {
-                if ($AddressFamily -ne "IPv6")
-                {
-                    throw "Server address *$_* is in IPv6 format, which does not match server address family $AddressFamily. Please correct either of them in the configuration and try again"
-                }
-             }
-         }
-    try
-    {        
-        Write-Verbose -Message "Checking the DNS Server Address ..."
-        #Get the current IP Address based on the parameters given.
-        $currentAddress = (Get-DnsClientServerAddress -InterfaceAlias $InterfaceAlias -AddressFamily $AddressFamily -ErrorAction Stop).ServerAddresses
-
-        #Check if the Server addresses are the same as the desired addresses.
-        if(@(Compare-Object -ReferenceObject $currentAddress -DifferenceObject $Address -SyncWindow 0).Length -gt 0)
+    if(-not (Get-NetAdapter | Where-Object -Property Name -EQ $InterfaceAlias ))
+    {
+                throw ( @(
+                    "Interface $InterfaceAlias is not available. "
+                    'Please select a valid interface and try again'
+                    ) -join '' )
+    }
+    if ($Address)
```

Comment:
Is there a situation where we are going to be passing in an address? If not we can move this entire thing up a level.